tests/test_pipeline/test_pipeline.py

```diff
@@ -356,3 +362,130 @@ def test_forecast_pipeline_with_nan_at_the_end(df_with_nans_in_tails):
     pipeline.fit(TSDataset(df_with_nans_in_tails, freq="1H"))
     forecast = pipeline.forecast()
     assert len(forecast.df) == 5
+
+
+@pytest.mark.parametrize(
+    "n_folds, mode, expected_masks",
+    (
+        (
+            2,
+            "expand",
+            [
+                FoldMask(
+                    first_train_timestamp="2020-01-01",
+                    last_train_timestamp="2020-04-03",
+                    target_timestamps=["2020-04-04", "2020-04-05", "2020-04-06"],
+                ),
+                FoldMask(
+                    first_train_timestamp="2020-01-01",
+                    last_train_timestamp="2020-04-06",
+                    target_timestamps=["2020-04-07", "2020-04-08", "2020-04-09"],
+                ),
+            ],
+        ),
+        (
+            2,
+            "constant",
+            [
+                FoldMask(
+                    first_train_timestamp="2020-01-01",
+                    last_train_timestamp="2020-04-03",
+                    target_timestamps=["2020-04-04", "2020-04-05", "2020-04-06"],
+                ),
+                FoldMask(
+                    first_train_timestamp="2020-01-04",
+                    last_train_timestamp="2020-04-06",
+                    target_timestamps=["2020-04-07", "2020-04-08", "2020-04-09"],
+                ),
+            ],
+        ),
+    ),
+)
+def test_generate_masks_from_n_folds(example_tsds: TSDataset, n_folds, mode, expected_masks):
+    masks = Pipeline._generate_masks_from_n_folds(ts=example_tsds, n_folds=n_folds, horizon=3, mode=mode)
+    for mask, expected_mask in zip(masks, expected_masks):
+        assert mask.first_train_timestamp == expected_mask.first_train_timestamp
+        assert mask.last_train_timestamp == expected_mask.last_train_timestamp
+        assert [timestamp for timestamp in mask.target_timestamps] == [
```

Comment:
Why do we need iteration here? Why can't we compare lists?

Reply:
Fixed, in the first version I casted the elements to timestamps)